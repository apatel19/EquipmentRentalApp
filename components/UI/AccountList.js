import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity, Platform} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const AccountList = props => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.touchableContainer}
        onPress={props.onPress}>
        <View style={styles.mainContainer}>
          <View style={styles.textContainer}>
            <Text style={styles.titleText}>{props.title}</Text>
            {props.subtitle && (
              <Text style={styles.subtitleText}>{props.subtitleText}</Text>
            )}
          </View>
          <View style={styles.iconContainer}>
            <Icon
              name={
                Platform.OS === 'android'
                  ? 'md-' + props.iconName
                  : 'ios-' + props.iconName
              }
              size={23}
              color={'black'}
            />
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 100,
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: 'black',
  },
  touchableContainer: {
    flex: 1,
    alignItems: 'center',
    padding: 10,
  },
  mainContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  textContainer: {
    flexDirection: 'column',
    width: '80%',
  },
  iconContainer: {
    width: '20%',
    alignItems: 'flex-end',
  },
  titleText: {
    fontSize: 20,
  },
  subtitleText: {
    fontSize: 16,
    paddingVertical: 10,
  },
});

export default AccountList;
