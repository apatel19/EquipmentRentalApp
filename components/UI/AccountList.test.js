import React from 'react';
import AccountList from './AccountList';
import renderer from 'react-test-renderer';

test('Card snapshot match test', () => {
  const tree = renderer.create(<AccountList />).toJSON();
  expect(tree).toMatchSnapshot();
});
