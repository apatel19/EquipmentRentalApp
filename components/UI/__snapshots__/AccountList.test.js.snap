// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot match test 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "height": 100,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "width": "20%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 23,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ?
        </Text>
      </View>
    </View>
  </View>
</View>
`;
